prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -510,6 +512,115 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         Ok(still_unresolved_edges)
     }
 
+    fn stmt_preconditions(&self, stmt: &vir::Stmt) -> Vec<vir::Expr> {
+        struct FindFnApps<'a, 'b, 'c>(Vec<vir::Expr>, &'a Encoder<'b, 'c>);
+        impl<'a, 'b, 'c> StmtWalker for FindFnApps<'a, 'b, 'c> {
+            fn walk_exhale(&mut self, statement: &vir::Exhale) {
+                self.0.push(statement.expr.clone());
+            }
+            fn walk_assert(&mut self, statement: &vir::Assert) {
+                self.0.push(statement.expr.clone());
+            }
+            fn walk_method_call(&mut self, statement: &vir::MethodCall) {
+                // Note: We know that in Prusti method's preconditions and postconditions are empty
+                for arg in &statement.arguments {
+                    self.walk_expr(arg);
+                }
+            }
+            fn walk_expr(&mut self, expr: &vir::Expr) {
```

Comment:
Why not use an `ExprWalker` here?

Reply:
Hmm I missed that since it's in a separate file compared to the definition of `Expr`, whereas the Statement one isn't